geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/KafkaConsumerFeatureSource.scala

```diff
@@ -123,9 +127,26 @@ trait KafkaConsumerFeatureCache extends QuadTreeFeatureStore {
     if (geometries.isEmpty) {
       unoptimized(a)
     } else {
-      val envelope = geometries.head.getEnvelopeInternal
-      geometries.tail.foreach(g => envelope.expandToInclude(g.getEnvelopeInternal))
-      new DFR(sft, new DFI(spatialIndex.query(envelope, a.evaluate)))
+      val set = geometries.toSet
+
+      if (set.size == 1) {
+        val envelope = set.head.getEnvelopeInternal
+        new DFR(sft, new DFI(spatialIndex.query(envelope, a.evaluate)))
+      } else if (set.size == 2 && !set.head.overlaps(set.tail.head)) {
+
+        val env1 = set.head.getEnvelopeInternal
+        val env2 = set.tail.head.getEnvelopeInternal
```

Comment:
Ah... this is a separate issue unrelated to geo-indexing + attribute-indexing in the KDS:).

I believe I wrote the first pass here...  we were trying to fix filters which cross the IDL.  Such filters have two geometries; that's why the case for two is hard-coded.  A more general solution requires sorting out of non-overlapping geometries.